### PR TITLE
use testcontainers spring context customizer for redis

### PR DESCRIPTION
**Pull request by Tcharl**
Aligns testcontainer bootstrap using spring contextcustomizer
---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

Closes the https://github.com/jhipster/generator-jhipster/issues/17691 saga

### CODE REVIEW

**Comment by DanielFran** [comment]
@Tcharl can you please fix daily builds prior?
See https://github.com/jhipster/generator-jhipster/pull/18513#issuecomment-1141777993

**Review by mshima — COMMENTED** [review]
Need to cleanup removed files.

**Comment by Tcharl** [comment]
Fixed, thank you for pointing it out!

File: package-lock.json [hunk 1]
```diff
@@ -5,7 +5,6 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "generator-jhipster",
       "version": "7.8.1",
```
Comment (mshima) [inline package-lock.json:8]:
```suggestion
      "name": "generator-jhipster",
      "version": "7.8.1",
```

Comment (Tcharl) [reply package-lock.json]: sry, npm link is annoying...

Comment (mshima) [reply package-lock.json]: Maybe related with old npm.

**Pull request merged**